includes/modules/analytics/google/class-request.php

```diff
@@ -136,6 +145,20 @@ public function http_delete( $url, $args = [], $timeout = 10 ) {
 	 * @return array|false Assoc array of decoded result.
 	 */
 	private function make_request( $http_verb, $url, $args = [], $timeout = 10 ) {
+		// Early Bail!!
+		if ( ! $this->refresh_token() && ! $this->is_notice_added ) {
```

Comment:
Not a good idea to call the function of child class from a parent.

The function is used in `RankMath\Google\Request` where is it is defined in `RankMath\Google\Api`. And the class structure is something like below ( Parent => Child ). And that can potentially cause an issue when any intermediate class call `RankMath\Google\Request::make_request()`.

`RankMath\Google\Request` => `RankMath\Google\Analytics` => `RankMath\Google\Console` => `RankMath\Google\Api`